### DISCUSSION TITLE
Raising the error if input model has extra parameters?

### DISCUSSION THREAD

**Discussion by rappongy** [fastapi/fastapi#8016] [Questions] [answered]
Hello

How can I raise the error if input pydantic's model has extra parameters?
Right now all extra keys are just ignored.

**Example:**
```
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class UpdateModel(BaseModel):
    name: str

@app.post("/items/")
async def update(item: UpdateModel):
    # do smth
```
If I send `{"name": "FOO", "id": "BAR"}` the ID key will be ignored. 
Is it any common way to make method to raise error in that case?

---
Thanks for your work.

**Comment by dmontagu** [comment, marked as answer]
I believe the following should cause this to raise errors:
```
class UpdateModel(BaseModel):
    name: str
    class Config:
        extra = "forbid"
```

This should be documented in the pydantic docs.

**Comment by rappongy** [comment]
Thank you! Missed this in docs.

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @rappongy  :+1:
